polygeist/lib/Dialect/Polygeist/Transforms/ParallelLower.cpp

```diff
@@ -183,10 +195,12 @@ mlir::LLVM::LLVMFuncOp GetOrCreateFreeFunction(ModuleOp module) {
           symbolTable.lookupSymbolIn(module, builder.getStringAttr("free"))))
     return fn;
   auto *ctx = module->getContext();
+  auto VoidPtrTy =
+      (useOpaquePointers)
+          ? LLVM::LLVMPointerType::get(ctx)
+          : LLVM::LLVMPointerType::get(mlir::IntegerType::get(ctx, 8));
```

Comment:
This code patterned is repeated. Makes me think we could have a utility function to create a pointer type based on whether we use opaque ptrs. Something like this:
```
static LLVM::LLVMPointerType getPtrType(Type elemTy, std::optional<unsigned> memSpace, MLIRContext ctx) {
  if (memSpace.has_value())
    return (useOpaquePointers)
            ? LLVM::LLVMPointerType::get(ctx, *memSpace)
            : LLVM::LLVMPointerType::get(elemTy, *memSpace);
    return (useOpaquePointers)
            ? LLVM::LLVMPointerType::get(ctx)
            : LLVM::LLVMPointerType::get(elemTy);
}
```
WDYT?

Reply:
Good idea, added that helper in the latest commit. 

As the other helpers in that file (`GetOrCreateMallocFunction` and `GetOrCreateFreeFunction`) are not static, I decided to not make it static. Let me know if you prefer it to be static.